### ISSUE TITLE
ContainSingle(predicate) throws FormatException

### ISSUE THREAD

**Issue by abatishchev** [fluentassertions/fluentassertions#1514]
### Description

`ContainSingle(Expression<Func<T, bool>>, string, params object[])` throws `FormatException` on assertion failure.

### Complete minimal example reproducing the issue

```c#
try
{
    await _sdkClient.Accounts.CreateAsync(resourceGroup, account.Name, account, cancellationToken),
}
catch (ErrorResponseException ex) when (ex.Response.StatusCode == HttpStatusCode.BadRequest)
{
    ex.Body.Error.Details
      .Should()
      .ContainSingle(d => d.Message == $"Account '{account.Name}' already exists" &&
                          d.Code == ErrorCode.AccountValidationFailed.Name);
}
```

I know this code won't compile but I struggle to get the exception reproduced locally. It occurs only on the build server during running the test.

### Expected behavior:

Assertion should either succeed or fail.

### Actual behavior:

```
Test method ResourceProviderService.Tests.Functional.AccountTests.Create_And_Delete_Duplicate_Accounts_Using_Sdk threw exception: 

System.FormatException: Index (zero based) must be greater than or equal to zero and less than the size of the argument list.

  Stack Trace:
   at System.Text.StringBuilder.AppendFormatHelper(IFormatProvider provider, String format, ParamsArray args)
   at System.String.FormatHelper(IFormatProvider provider, String format, ParamsArray args)
   at System.String.Format(String format, Object[] args)
   at FluentAssertions.Execution.MessageBuilder.FormatArgumentPlaceholders(String failureMessage, Object[] failureArgs)
   at FluentAssertions.Execution.MessageBuilder.Build(String message, Object[] messageArgs, String reason, ContextDataItems contextData, String identifier, String fallbackIdentifier)
   at FluentAssertions.Execution.AssertionScope.<>c__DisplayClass30_0.<FailWith>b__0()
   at FluentAssertions.Execution.AssertionScope.FailWith(Func`1 failReasonFunc)
   at FluentAssertions.Execution.AssertionScope.FailWith(Func`1 failReasonFunc)
   at FluentAssertions.Execution.AssertionScope.FailWith(String message, Object[] args)
   at FluentAssertions.Collections.SelfReferencingCollectionAssertions`2.ContainSingle(Expression`1 predicate, String because, Object[] becauseArgs)
   at ResourceProviderService.Tests.Functional.AccountTests.Create_And_Delete_Duplicate_Accounts_Using_Sdk(String resourceGroup, Account account) in D:\a\1\s\src\ResourceProviderService.Tests.Functional\AccountTests.cs:line 148
   at Microsoft.VisualStudio.TestPlatform.MSTestAdapter.PlatformServices.ThreadOperations.ExecuteWithAbortSafety(Action action)
```

### Versions

* Which version of Fluent Assertions are you using?

```xml
<PackageReference Include="FluentAssertions" Version="5.10.3" />
```

* Which .NET runtime and version are you targeting?

.NET Core 3.1 + .NET 5 (on hosted pool in Azure DevOps / Pipelines)

**Comment by dennisdoomen** [comment]
> .NET Core 3.1 + .NET 5 (on hosted pool in Azure DevOps / Pipelines)

What does that mean? Is your build server running the code against a different .NET runtime than the one you use during development?

**Comment by jnyrup** [comment]
This looks like a duplicate of #1404, which was fixed in #1406 and is part of 6.0.0-alpha2.
Please try if 6.0.0-alpha2 solves your issue.

Alternatively the workaround should be to do the string interpolation outside the predicate.
```cs
var message = $"Account '{account.Name}' already exists";
ex.Body.Error.Details
      .Should()
      .ContainSingle(d => d.Message == message &&
                          d.Code == ErrorCode.AccountValidationFailed.Name);
```

**Comment by abatishchev** [comment]
@dennisdoomen,

> What does that mean?

The Hosted Pool in ADO is managed by ADO so customers don't have a say on what software is installed. The right term for if would be "Fully Manager" perhaps. Has its own pros and cons (I don't need to worry and install or update anything).

So the tests run on a VM having both 3.1 and 5 installed. They target 3.1 though.

@jnyrup thanks! I'll give it a try. Here's my workaround for now:

```c#
var details = ex.Body.Error.Details
                .Should()
                .ContainSingle()
                .Which;
details.ErrorCode.Should().Be(ErrorCode.AccountValidationFailed.Name);
details.Message.Should().Be(message);
```
